acceptance/new-user/sign-up.test.js

```diff
@@ -1,105 +1,115 @@
-const puppeteer = require('puppeteer');
-const config = require('config');
+const puppeteer = require("puppeteer");
+const config = require("config");
 
-const URLS = require('../../constants/urls');
+const URLS = require("../../constants/urls");
 
 const { HOME_PAGE, SIGN_UP_PAGE, MY_HOST } = URLS;
 
 let browser, page;
 
+function delay(time) {
+    return new Promise(function (res) {
+        setTimeout(res, time);
+    });
+}
+
 beforeAll(async () => {
-  browser = await puppeteer.launch({
-    // headless: false,
-    // slowMo: 250
-  });
-
-  const context = await browser.createIncognitoBrowserContext();
-
-  page = await context.newPage();
-  // await page.emulate(puppeteer.devices['iPhone 8']);
-  // await page.setViewport({
-  //   width: 1440,
-  //   height: 900,
-  //   deviceScaleFactor: 1,
-  // });
-  return page;
+    browser = await puppeteer.launch({
+        headless: false,
+        slowMo: 0,
+    });
+
+    const context = await browser.createIncognitoBrowserContext();
+
+    page = await context.newPage();
+    // await page.emulate(puppeteer.devices['iPhone 8']);
+    // await page.setViewport({
+    //   width: 1440,
+    //   height: 900,
+    //   deviceScaleFactor: 1,
+    // });
+    return page;
 });
 
 afterAll(async () => {
-  await browser.close();
+    await browser.close();
 });
 
+describe("New user navigates", () => {
+    test("New user sess a sign up page", async () => {
+        await page.goto(HOME_PAGE);
+
+        await page.waitForResponse((res) => res.url().endsWith("/users/self"));
+        await page.waitForSelector("button.login-btn-text");
+
+        await Promise.all([
+            page.waitForNavigation(),
+            page.click("button.login-btn-text"),
+        ]);
+
+        const ghUsernameInput = await page.waitForSelector("input#login_field");
```

Comment:
_Doubt:_
What is `gh` means here in the identifier name `ghUsernameInput`?

Also, the `id` has a BEM naming convention ?

Reply:
I honestly don't know this one.. It was already there, added by Ankush, so I didn't touch it 😅. About the BEM convention, this input field is of the github sign in form, which appears when a new user signs up for our website using his/her github account. So we are basically selecting that field and typing our test github username and password through puppeteer. The classnames in this particular form follow the BEM convention and that's why we have to write like that.

Reply:
Meaning it is a field which is `fetched` not `created` by the us (tester) ?
Am I understanding this correctly ?